src/vehicle.cpp

```diff
@@ -3756,9 +3764,35 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop(){
+    std::vector<int> scoops=all_parts_with_feature("SCOOP");
+    auto c=get_points();
+    for(int scoop:scoops){
+        auto position = global_pos3()+parts[scoop].precalc[0];
+        const char *sound_msgs[]={"Whirrrr","Ker-chunk","Swish","Cugugugugug","Horrifying"};
+        sounds::sound(position, (int) rng(20,35), sound_msgs[rng(0,3)]);
+        for(tripoint i:c){
+            if(g->m.moppable_items_at(i)){//That's right, street sweepers can now clean streets
+                g->m.mop_spills(i);
+                continue;//the item should not be accessed at this point.
+            }
+            if( g->m.has_items(i) ){
+                item* that_item_there = g->m.item_from(i,0);//remove the first item on each square.
```

Comment:
There should be some check for validity of scooping an item. Scooping a whole corpse should not be OK, even if it can't be contained.

Also, battery should be discharged before scooping and the scoop should fail if that discharge wasn't enough to lift the item up.